### ISSUE TITLE
SassError: This file is already being loaded (when using additionalData)

### ISSUE THREAD

**Issue by matttk** [webpack-contrib/sass-loader#908]
- Operating System: Windows 10
- Node Version: 12.14.1
- NPM Version: 6.14.8
- webpack Version: 5.10.3
- sass-loader Version: 10.1.0

### Expected Behavior

`additionalData` should work like `prependData`.

### Actual Behavior

`additionalData` causes compile error (I think).

```ERROR in ./scss/themes/default.scss
Module build failed (from ../node_modules/mini-css-extract-plugin/dist/loader.js):
ModuleBuildError: Module build failed (from ../node_modules/sass-loader/dist/cjs.js):
SassError: This file is already being loaded.
  ╷
1 │ @import "app/scss/themes/default.scss";
  │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ╵
  app\scss\themes\default.scss 1:9  root stylesheet
```

### Code

```js
// webpack.config.js
{
          test: /\.(css|scss)$/,
          use: [
            MiniCssExtractPlugin.loader,
            {
              loader: "css-loader",
              options: { modules: false, importLoaders: 1 }
            },
            "postcss-loader",
            {
              loader: "sass-loader",
              options: {
                additionalData: `@import "app/scss/themes/${theme}.scss";`
              }
            }
          ]
        }
```

### How Do We Reproduce?

Well, I hope you only need to use the above code, more or less. I've been working through a webpack 4 to 5 upgrade and followed the advice to upgrade all loaders. I used to use `prependData` to prepend a theme file `@import` to all my `SCSS` files. After upgrading, I noticed that has changed to `additionalData` but I can't get it to work. I get the above error, that the file has already been imported. I can't rule out there is another cause but I also don't know what should be so complicated here. Am I overlooking something or is there an error with `additionalData`?

**Comment by alexander-akait** [comment]
This error from sass, it means `@import "app/scss/themes/${theme}.scss";` already exists in your code, sorry I can't say where, because you don't provide your code, feel free to feedback

**Comment by matttk** [comment]
> This error from sass, it means `@import "app/scss/themes/${theme}.scss";` already exists in your code, sorry I can't say where, because you don't provide your code, feel free to feedback

I do not import this file anywhere in my code ,other than on this one line in webpack.config.

**Comment by matttk** [comment]
The only other place where the file is mentioned is also in webpack:

```
new webpack.ProvidePlugin({
  THEME_VARS: path.resolve(__dirname, `app/scss/themes/${theme}.scss`)
}),```

But this is unrelated, as far as I understand.

**Comment by alexander-akait** [comment]
I need to look at code, this error from `sass`/`node-sass`, there is no problems/errors from the loader

**Comment by matttk** [comment]
What code do you need to see? I cannot post the entire codebase here because it is private. However, the code is also completely unchanged from before. The only thing that changed is that I upgraded `webpack`, `sass-loader`, and some other loaders and plugins, wihch also meant I needed to update `webpack.config.js` and needed to change `prependData` to `additionalData`.

**Comment by alexander-akait** [comment]
@matttk Maybe small reproducible example? If `sass` say what we have duplicate import, it means we have it (or bug on `sass` side), but if we need to know where this is happening, we should look at code (minimum example), I cannot fix or solve the problem the cause of which I cannot understand

**Comment by alexander-akait** [comment]
Unfortunately sass doesn't provide good stack trace for errors, maybe it helps to solve these problems faster

**Comment by matttk** [comment]
OK, I forked and updated a [sandbox](https://codesandbox.io/s/webpack-config-forked-mft96) and can't reproduce the problem. More importantly, I did the obvious thing I didn't think of yesterday and downgraded `sass-loader` to `8.0.2` and put back `prependData`. I still have the problem.... so now I have no clue where to look but at least you are right to close the issue here. 😓

Thanks anyway!

**Comment by alexander-akait** [comment]
You can debug `sass` (`dast-sass`) (they have transpiled code from dart to JS, but it is still readable), just look where you already import this

**Comment by matttk** [comment]
Just wanted to give a quick new year update - I don't know why I didn't think of this but I downgraded to 8.0.2 and then switched back to `prependData`. Now my build is successful. 🤷‍♂️

**Comment by matttk** [comment]
Maybe I spoke too soon because now I realised no scss files are being imported in js files anymore. :/

**Comment by matttk** [comment]
OK, I changed 2 things and now I don't get the error anymore. The first was, as I mentioned above, that I got rid of `MiniCssExtractPlugin`. The second was that I found [this stackoverflow](https://stackoverflow.com/questions/61517117/cannot-export-sass-variables-into-javascript) and tried separating the SCSS variables from the `:export` in 2 different files. Later in my code, I use `ProvidePlugin` to set the variables from the theme file to a global variable and I thought maybe this is the duplicate import.

I re-upgraded to the latest version of `sass-loader` and the build works now. The interesting thing is the above 2 things I fixed were not problems for 8.0.2.

Now I only have the problem that `ProvidePlugin` imports nothing from the exported variables in the SCSS file but this is a different issue and I'm not sure yet if it's related to `sass-loader`.

**Comment by takhello** [comment]
The same problem arises， 
version：   "sass-loader": "^10.1.1",
tip:
Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
SassError: This file is already being loaded.

Issue fixed after demotion：
 "sass-loader": "^8.0.2",

**Comment by alexander-akait** [comment]
@takhello please create reproducible test repo (and open a new issue)

**Comment by demetranadya** [comment]
> The same problem arises， version： "sass-loader": "^10.1.1", tip: Module build failed (from ./node_modules/sass-loader/dist/cjs.js): SassError: This file is already being loaded.
> 
> Issue fixed after demotion： "sass-loader": "^8.0.2",

Yes, then need change additionData on prependData
